File: project-euler.java/src/main/java/euler/solutions/_20.java
package euler.solutions;

import euler.common.BaseSolution;
import euler.common.Result;

import java.math.BigInteger;

public class _20 extends BaseSolution {

    @Override
    public Result solve() {
        BigInteger num = new BigInteger("1");

        for (int i = 2; i <= 100; i++) {
            num = num.multiply(new BigInteger(String.valueOf(i)));
        }

        int sum = 0;
        String numStr = String.valueOf(num);
        for (int i = 0; i < numStr.length(); i++) {
            char chr = numStr.charAt(i);
            sum += chr - 48;
        }

        return new Result(sum);
    }
}

File: project-euler.go/solutions/_20/20.go
package _20

import "math/big"

func Solve() int64 {
	var num big.Int
	num.SetInt64(1)

	for i := 2; i <= 100; i++ {
		var iBig big.Int
		iBig.SetInt64(int64(i))
		num.Mul(&num, &iBig)
	}

	sum    := 0
	numStr := num.String()
	for i := range numStr {
		sum += int(numStr[i]) - 48
	}

	return int64(sum)
}
